x-pack/plugins/lens/public/react_embeddable/renderer/lens_custom_renderer_component.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ReactEmbeddableRenderer } from '@kbn/embeddable-plugin/public';
+import { useSearchApi } from '@kbn/presentation-publishing';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import type { LensApi, LensRendererProps, LensRuntimeState, LensSerializedState } from '../types';
+import { LENS_EMBEDDABLE_TYPE } from '../../../common/constants';
+import { createEmptyLensState } from '../helper';
+
+/**
+ * The aim of this component is to provide a wrapper for other plugins who want to
+ * use a Lens component into their own page. This hides the embeddable parts of it
+ * by wrapping it into a ReactEmbeddableRenderer component and exposing a custom API
+ */
+export function LensRenderer({
+  title,
+  withDefaultActions,
+  extraActions,
+  showInspector,
+  syncColors,
+  syncCursor,
+  syncTooltips,
+  viewMode,
+  id,
+  ...props
+}: LensRendererProps) {
+  // Use the settings interface to store panel settings
+  const settings = useMemo(() => {
+    return {
+      syncColors$: new BehaviorSubject(false),
+      syncCursor$: new BehaviorSubject(false),
+      syncTooltips$: new BehaviorSubject(false),
+    };
+  }, []);
+
+  // view mode needs to be a BehaviorSubject
+  const viewMode$Ref = useRef(new BehaviorSubject(viewMode));
+  if (viewMode !== viewMode$Ref.current.getValue()) {
+    viewMode$Ref.current.next(viewMode);
+  }
+
+  // Lens API will be set once, but when set trigger a reflow to adopt the latest attributes
+  const [lensApi, setLensApi] = useState<LensApi | undefined>(undefined);
```

Comment:
I'm using `useState` vs `useRef` to prevent stale attributes to stick if they are pushed quicker than the `lensApi` is built.
With this it forces a reflow with the latest version of the `attributes`.